utils/dependency_walker.go

```diff
@@ -0,0 +1,223 @@
+package utils
+
+import (
+	"fmt"
+
+	"github.com/VirusTotal/gyp/ast"
+)
+
+var yaraModules = []string{"pe", "elf", "cuckoo", "magic", "hash", "math", "dotnet", "time", "dex"}
+
+type queueT struct {
+	node       ast.Node
+	ignoreList []string
+}
+
+// GetDependencyChainForRules will recursively find dependencies for a list of rules
+// returnDepAsPrivate will dictate if dependent rules are marked as private
+func GetDependencyChainForRules(ruleset ast.RuleSet, ruleNames ...string) (ast.RuleSet, error) {
+	queue := make(map[string]struct{}) // List of rules to get dependencies for
```

Comment:
> It is because you don't want to insert duplicate items in the queue?

Yes I don't want to insert duplicate items - I'll rename this to pendingRules.